Content.Server/Explosion/EntitySystems/ExplosionSystem.TileFill.cs

```diff
@@ -0,0 +1,330 @@
+            // cannot break. if the remaining intensity remains unchanged TWO loops in a row, we know that this is the
+            // case.
+            if (intensityUnchangedLastLoop && remainingIntensity == previousIntensity)
+                break;
+
+            intensityUnchangedLastLoop = remainingIntensity == previousIntensity;
+            iteration += 1;
+        }
+
+        // Neighbor finding is done. Perform final clean up and return.
+        foreach (var grid in gridData.Values)
+        {
+            grid.CleanUp();
+        }
+        spaceData?.CleanUp();
+
+        return (totalTiles, iterationIntensity, spaceData, gridData, spaceMatrix);
+    }
+
+    /// <summary>
+    ///     Look for grids in an area and returns them. Also selects a special grid that will be used to determine the
+    ///     orientation of an explosion in space.
+    /// </summary>
+    /// <remarks>
+    ///     Note that even though an explosion may start ON a grid, the explosion in space may still be orientated to
+    ///     match a separate grid. This is done so that if you have something like a tiny suicide-bomb shuttle exploding
+    ///     near a large station, the explosion will still orient to match the station, not the tiny shuttle.
+    /// </remarks>
+    public (List<GridId>, GridId?) GetLocalGrids(MapCoordinates epicenter, float totalIntensity, float slope, float maxIntensity)
+    {
+        // Get the explosion radius (approx radius if it were in open-space). Note that if the explosion is confined in
+        // some directions but not in others, the actual explosion may reach further than this distance from the
+        // epicenter. Conversely, it might go nowhere near as far.
+        var radius = 0.5f + IntensityToRadius(totalIntensity, slope, maxIntensity);
+
+        // to avoid a silly lookup for silly input numbers, cap the radius to half of the theoretical maximum (lookup area gets doubled later on).
+        radius = Math.Min(radius, MaxIterations / 4);
+
+        GridId? referenceGrid = null;
+        float mass = 0;
+
+        // First attempt to find a grid that is relatively close to the explosion's center. Instead of looking in a
+        // diameter x diameter sized box, use a smaller box with radius sized sides:
+        var box = Box2.CenteredAround(epicenter.Position, (radius, radius));
+
+        foreach (var grid in _mapManager.FindGridsIntersecting(epicenter.MapId, box))
+        {
+            if (TryComp(grid.GridEntityId, out PhysicsComponent? physics) && physics.Mass > mass)
+            {
+                mass = physics.Mass;
+                referenceGrid = grid.Index;
+            }
+        }
+
+        // Next, we use a much larger lookup to determine all grids relevant to the explosion. This is used to determine
+        // what grids should be includes during the grid-edge transformation steps. This means that if a grid is not in
```

Comment:
```suggestion
        // what grids should be included during the grid-edge transformation steps. This means that if a grid is not in
```

There was also a ctrl-H typo somewhere that I forgot to comment.